core/src/main/java/org/elasticsearch/index/query/DisMaxQueryBuilder.java

```diff
@@ -54,6 +61,24 @@ public DisMaxQueryBuilder add(QueryBuilder queryBuilder) {
     }
 
     /**
+     * Add a list of sub-queries to this disjunction.
+     */
+    public DisMaxQueryBuilder add(List<QueryBuilder> queryBuilders) {
+        for (QueryBuilder<?> queryBuilder : queryBuilders) {
+            add(queryBuilder);
+        }
+        return this;
+    }
+
+
+    /**
+     * @return an immutable list copy of the current sub-queries of this disjunction
+     */
+    public List<QueryBuilder> queries() {
+        return ImmutableList.copyOf(this.queries);
```

Comment:
curious about making a copy here. Did we do this in other queries too? You went this path to prevent users from making changes to the list that we might return instead?